packages/core/src/Keymap.ts

```diff
@@ -0,0 +1,101 @@
+import { CommandIdentifier, CommandArgs } from './Dispatcher';
+
+interface ShortcutPattern {
+    key: string;
+    modifiers: Set<string>;
+}
+// todo: should I place this in file Command?
+export interface CommandCall {
+    commandId: string;
+    commandArgs?: CommandArgs;
+}
+
+interface ShortcutItem extends CommandCall {
+    pattern: ShortcutPattern;
+    priority: number;
+}
+
+// todo: questions
+// - who preventDefault / stopPropagation / stopImmediatePropagation?
+// - what about when a pattern is already present in the system?
+//   - if the check (static or dynamic) is different?
+export class Keymap {
+    _shortcutItems: ShortcutItem[] = [];
+
+    //--------------------------------------------------------------------------
+    // public
+    //--------------------------------------------------------------------------
+
+    addShortuct(
+        patternString: string,
+        commandId: CommandIdentifier,
+        commandArgs?: CommandArgs,
+        priority = 0,
+    ): void {
```

Comment:
Can't we take a `Shortcut` object ? You know, the one from `JWPlugin` that I would like to see here... :p

Reply:
The `Shortcut` object has a `platform` property. Keymap should not know about the platform.
And here, the first argument is a pattern as string so it cannot have a `ShortcutItem`.
What do you suggest? Another interface for the function ?